src/Pages/Account/Account.js

```diff
@@ -0,0 +1,421 @@
+import React from "react";
+import { SERVER, SIGNUP_API, SIGNIN_API } from "../../config";
+import { Link } from "react-router-dom";
+import "./Account.scss";
+
+class Account extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      firstname: "",
+      firstnamehasValue: true,
+      lastname: "",
+      lastnamehasValue: true,
+      email: "",
+      password: "",
+      passwordhasValue: true,
+      checkPw: "",
+      checkPwhasValue: true,
+      adagreement: false,
+      serviceagreement: false,
+      serviceagreementhasValue: true,
+    };
+  }
+
+  // 인풋의 변화연동
+  firstnameInput = (e) => {
+    const firstnameValue = e.target.value;
+    this.setState(
+      {
+        firstname: firstnameValue,
+      },
+      () =>
+        this.setState({
+          firstnamehasValue: this.state.firstname.length > 0 ? true : false,
+        }),
+    );
+  };
+
+  lastnameInput = (e) => {
+    const lastnameValue = e.target.value;
+    this.setState({ lastname: lastnameValue }, () =>
+      this.setState({
+        lastnamehasValue: this.state.lastname.length > 0 ? true : false,
+      }),
+    );
+  };
+
+  emailInput = (e) => {
+    const emailValue = e.target.value;
+    this.setState({ email: emailValue }, () =>
+      this.setState({
+        emailhasValue: this.state.email.length > 0 ? true : false,
+      }),
+    );
+  };
+
+  passwordInput = (e) => {
+    const passwordValue = e.target.value;
+    this.setState({ password: passwordValue }, () =>
+      this.setState({
+        passwordhasValue: this.state.password.length > 0 ? true : false,
+      }),
+    );
+  };
+
+  checkPwInput = (e) => {
+    const checkPwValue = e.target.value;
+    this.setState({ checkPw: checkPwValue }, () =>
+      this.setState({
+        checkPwhasValue: this.state.checkPw.length > 0 ? true : false,
+      }),
+    );
+  };
+
+  adagreementInput = (e) => {
+    const adagreementValue = e.target.checked;
+    this.setState({ adagreement: adagreementValue });
+  };
+
+  seviceagreementInput = (e) => {
+    const serviceagreementValue = e.target.checked;
+    this.setState({ serviceagreement: serviceagreementValue }, () =>
+      this.setState({
+        serviceagreementhasValue: this.state.serviceagreement ? true : false,
+      }),
+    );
+  };
```

Comment:
```suggestion
  seviceagreementInput = (e) => {
    const { checked } = e.target;
    this.setState({
      serviceagreement: checked,
      serviceagreementhasValue: checked,
    });
  };
```
- 위와 마찬가지로 구조 분해 할당 적용해주세요.
- `serviceagreement`, `serviceagreementhasValue` 같은 값을 가지므로 한 번에 setState 해주면 됩니다.
- 마찬가지로 이미 boolean 데이터 타입이기 때문에 삼항 연산자를 적용할 필요 없습니다.

Reply:
수정했습니다 !